packages/bun/src/BunAdapter.ts

```diff
@@ -0,0 +1,283 @@
+import { file } from 'bun';
+
+/**
+ * Bun route handler type
+ */
+type BunHandler = (request: Request) => Response | Promise<Response>;
+
+/**
+ * Bun routes object structure
+ * { [path: string]: { [method: string]: BunHandler } }
+ */
+type BunRoutes = Record<string, Record<string, BunHandler>>;
+
+export class BunAdapter implements IServerAdapter {
+  private basePath = '/';
+  private bullBoardQueues: BullBoardQueues | undefined;
+  private errorHandler: ((error: Error) => ControllerHandlerReturnType) | undefined;
+  private uiConfig: UIConfig = {};
+  private staticRoute?: string;
+  private staticPath?: string;
+  private viewPath?: string;
+  private entryRoute?: AppViewRoute;
+  private apiRoutes: AppControllerRoute[] = [];
+
+  public setBasePath(path: string): this {
+    this.basePath = path;
+    return this;
+  }
+
+  public setStaticPath(staticRoute: string, staticPath: string): this {
+    this.staticRoute = staticRoute;
+    this.staticPath = staticPath;
+    return this;
+  }
+
+  public setViewsPath(viewPath: string): this {
+    this.viewPath = viewPath;
+    return this;
+  }
+
+  public setErrorHandler(handler: (error: Error) => ControllerHandlerReturnType): this {
+    this.errorHandler = handler;
+    return this;
+  }
+
+  public setApiRoutes(routes: readonly AppControllerRoute[]): this {
+    this.apiRoutes = [...routes];
+    return this;
+  }
+
+  public setEntryRoute(routeDef: AppViewRoute): this {
+    this.entryRoute = routeDef;
+    return this;
+  }
+
+  public setQueues(bullBoardQueues: BullBoardQueues): this {
+    this.bullBoardQueues = bullBoardQueues;
+    return this;
+  }
+
+  public setUIConfig(config: UIConfig = {}): this {
+    this.uiConfig = config;
+    return this;
+  }
+
+  /**
+   * Builds and returns all routes as a Bun routes object
+   * Returns an object where keys are paths and values are { METHOD: handler }
+   */
+  public getRoutes(): BunRoutes {
+    if (!this.staticRoute || !this.staticPath) {
+      throw new Error(`Please call 'setStaticPath' before using 'getRoutes'`);
+    }
+    if (!this.entryRoute) {
+      throw new Error(`Please call 'setEntryRoute' before using 'getRoutes'`);
+    }
+    if (!this.viewPath) {
+      throw new Error(`Please call 'setViewsPath' before using 'getRoutes'`);
+    }
+    if (!this.uiConfig) {
+      throw new Error(`Please call 'setUIConfig' before using 'getRoutes'`);
+    }
+    if (!this.bullBoardQueues) {
+      throw new Error(`Please call 'setQueues' before using 'getRoutes'`);
+    }
+
+    const routes: BunRoutes = {};
+
+    // Register static file routes with wildcard
+    const staticBasePath = this.joinPaths(this.basePath, this.staticRoute);
+    routes[`${staticBasePath}/*`] = {
+      GET: async (request: Request) => {
+        const url = new URL(request.url);
+        const pathname = url.pathname;
+        const relativePath = pathname.replace(staticBasePath, '');
+        const fullPath = `${this.staticPath}${relativePath}`;

```

Comment:
This should be normilized to check if the path is still inside `this.staticPath`, so if an attacker crafts `/static/js/../../../.env`, it won't work